plugins/inputs/intel_pmt/README.md

```diff
@@ -0,0 +1,434 @@
+# Intel® Platform Monitoring Technology (Intel® PMT) Input Plugin
+
+Intel® Platform Monitoring Technology Input Plugin collects metrics from
+the Telemetry capability of the Intel® PMT kernel driver.
+
+Intel® Platform Monitoring Technology (Intel® PMT) is an architecture for
+enumerating and accessing hardware monitoring capabilities on a device.
+
+Intel PMT support has been added to the mainline Linux kernel
+under the platform driver (`drivers/platform/x86/intel/pmt`).
+This driver exposes the Intel PMT telemetry space as a sysfs entry
+at `/sys/class/intel_pmt/`.
+
+Each discovered Intel PMT telemetry aggregator is exposed as a directory named
+with a `telem` prefix. Every directory contains a file named `guid` identifying
+the unique PMT space. This file is associated with a set of XML specification
+files.
+
+Intel PMT XML files are available at [Intel-PMT Repository].
+
+Intel PMT Input Plugin discovers available PMT spaces and loads associated
+sets of PMT XML files. Using the specification inside the XML files,
+the plugin then reads low level samples/counters and evaluates high level
+samples/counters according to transformation formulas, and then reports the
+collected values.
+
+## Global configuration options <!-- @/docs/includes/plugin_config.md -->
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
+
+## Requirements
+
+Intel PMT Input Plugin requires an XML specification as `pmt_source`.
+It can be provided in two different ways:
+
+- as a URL – the plugin will attempt to download XML files.
+It will try to download the files again at every initialization
+and the files can be megabytes in size.
+
+- as a filepath – the plugin will use local copies.
+The provided filepath should be an absolute path to `pmt.xml` within
+local copies of XML files from the cloned [Intel-PMT Repository].
+
+## Configuration
+
+```toml @sample.conf
+# Intel Platform Monitoring Technology plugin exposes Intel PMT metrics available through the Intel PMT kernel space.
+# This plugin ONLY supports Linux.
+[[inputs.intel_pmt]]
+  ## URL or filepath to PMT XML within PMT repository.
+  ## Both URL and filepath should be absolute.
+  pmt_source = "https://raw.githubusercontent.com/intel/Intel-PMT/master/xml/pmt.xml"
```

Comment:
-1 on using a URL. I want local files only:

We are already running into issues with other plugins where someone added the ability to pull down from a URL and then as users start using this, ask for more and more of the functionality that essentially already exists in the common http client (e.g. https support, auth support, etc.)

I also do not like that we are advertising pulling down directly from github. While this is entirely possible, for a tool that can get deployed to thousands of nodes we are not a good citizen to suggest this.

Reply:
We can remove the URL as a source.

Reply:
URL is no longer an option